### PR TITLE
HIP: fix peer to peer memcopy

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Fix HIP p2p memcopy between devices within a single host system.
This PR additionally refactored the CUDA p2p implementation. 
For a long time CUDA is not requiring explicit p2p copy calls, `cudaMemcpy*` is detecting if a copy operation must be performed between different devices and will to all required configurations.

note: To test p2p memcpy #1399 is required too. Never the less this PR can be merged before #1399.

### CODE REVIEW

File: include/alpaka/mem/buf/uniformCudaHip/Copy.hpp [hunk 1]
```diff
@@ -364,13 +250,14 @@ namespace alpaka
...
-            Idx m_srcWidth; // required for 3D peer copy
+            Idx m_dstWidth;
+            Idx m_srcWidth;
 
             Idx m_extentHeight;
-            Idx m_dstHeight; // required for 3D peer copy
-            Idx m_srcHeight; // required for 3D peer copy
-
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+            Idx m_dstHeight;
```
Comment (j-stephan) [inline include/alpaka/mem/buf/uniformCudaHip/Copy.hpp:258]: Is there a reason we're not putting the width here, too? Also below for the 3D case.

Comment (psychocoderHPC) [reply include/alpaka/mem/buf/uniformCudaHip/Copy.hpp:258]: width is used somewhere in a call. All attributes which are guarded by the define are only active for debugging but are not required for any api call.

**Comment by psychocoderHPC** [comment]
@j-stephan Can we merge this PR?

**Review by sbastrakov — APPROVED** [review]

**Comment by j-stephan** [comment]
Yes, sorry for the delay. I somehow missed your response to my review.

**Pull request merged**